### PR TITLE
💎(site) Subnav updates

### LINKED ISSUE #217: Module: Resources
Design: https://www.figma.com/file/xtBKZcRBo7H2Jrjs7DjNqE/Final-Designs?node-id=2%3A4714

### PR DESCRIPTION
**Pull request by nickalcantara**
Updates styling of the subnav section and cards.

- Resolves #217

### CODE REVIEW

File: site/_components/card-subnav.njk [hunk 1]
```diff
@@ -4,7 +4,9 @@
   @param {object} navSection - Object containing the following attributes.
   @param {string} navSection.title - Nav section title.
   @param {string} navSection.desc - Nav section description.
-  @param {string} navSection.icon - Nav section icon.
+  @param {object} navSection.icon (optional) - Object containing the following attributes.
+  @param {string} navSection.icon.name - ID of icon within spritesheet.
+  @param {('primary-blue'|'primary-yellow'|'primary-green')} navSection.icon.background - Color keyword to apply as the icon's background circle.
```
Comment (Snugug) [inline site/_components/card-subnav.njk:9]: For simplicity's sake, we should make this just `blue`, `yellow`, or `green`. We can interpolate what the color should be from those, and we don't expose our CSS variables to our config, letting what particular shade we use be changeable without coupling it to our invocations or CSS variables directly

File: site/sass/components/_card-subnav.scss [hunk 2]
```diff
@@ -5,43 +5,79 @@
...
+      background-color: var(--primary-blue);
+    }
+
+    &--primary-yellow {
+      background-color: var(--primary-yellow);
+    }
+
+    &--primary-green {
+      background-color: var(--primary-green);
+    }
```
Comment (Snugug) [inline site/sass/components/_card-subnav.scss:73]: Same comments from the component itself; let's not couple these directly to the CSS var name

File: site/_components/item-grid.njk [hunk 3]
```diff
@@ -17,7 +17,10 @@
  #}-->
 {% from 'external-community.njk' import externalCommunity %} {% from 'card-subnav.njk' import cardSubnav %} {% from 'icon-card.njk' import iconCard %} {% from 'card.njk' import card %} {% macro itemGrid(title, items, modifiers, locale, defaultText) %} {% if items.length > 0 %}
 <section class="item-grid {{modifiers.custom}} {{'item-grid--icon-card' if modifiers.variant == 'icon'}} {{ 'item-grid--subnav' if modifiers.variant == 'subnav' }} {{'item-grid--community' if modifiers.variant == 'community'}} {{'item-grid--circled' if modifiers.shapeOn }}">
-  <h3 class="item-grid__title type--h3">{{title}}</h3>
+  <h3 class="item-grid__title type--h1">{{title}}</h3>
+  {% if modifiers.subtitle %}
+  <p class="item-grid__subtitle type--large">{{modifiers.subtitle}}</p>
+  {% endif %}
   <div class="item-grid__content">{% for item in items %} {% if modifiers.variant == 'icon' %} {{iconCard(item, defaultText)}} {% elif modifiers.variant == 'subnav' %} {% if item.icon and item.desc %} {{cardSubnav(item, locale)}} {% endif %} {% elif modifiers.variant == 'community' %} {{externalCommunity(item.logo, item.cta)}} {% else %} {{ card(item) }} {% endif %} {% endfor %}</div>
```
Comment (Snugug) [inline site/_components/item-grid.njk:24]: This also looks different than some of the other changes at this point that I've seen; is there a canonical one I should look at?

**Review by Snugug — CHANGES_REQUESTED** [review]
Thanks again for this work! It's looking great!

A couple of small comments below, plus there's one bigger one that I'd like you to implement: subgrid.
When all of the descriptor text in `card-subnav` are the same number of lines, everything lines up nicely, but when they're not, the design breaks a bit. A cross-browser push this year, known as [Interop 2022](https://web.dev/interop-2022/), is aiming to get a number of features launched and working across all browser. This includes [`subgrid`](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Subgrid), which should be able to fix this issue. It should work in Firefox right now, and then once other browsers gain support, should "magically" make it look great in those browsers, too.

Thanks!

File: site/_components/item-grid.njk (hunk 3 above)
Comment (nickalcantara) [reply site/_components/item-grid.njk:24]: Apologies for that - there were multiple minor changes that ultimately should have warranted a PR of its own. Once pieced together, [this is the desired outcome](https://gist.github.com/nickalcantara/51ef2e70d9f1e6edfca23e3e38a03081), though there would also be additions for inner item components as needed.

**Review by Snugug — APPROVED** [review]
Ooo I love how the subgrid looks in FF! Can't wait for Interop to land and we get that everywhere!

Thanks for your work on this!

**Pull request merged**